### PR TITLE
[Components] Gmail: Update find-email action to v0.1.2

### PR DESCRIPTION
**Pull request by hubertpysklo**
**Summary**
Adds email header extraction when `withTextPayload` is enabled to provide essential email metadata alongside the text content.

**Changes**
* **find-email.mjs**
   * Bumped version to `0.1.2`
   * **Moved** `message.message_id` extraction to only occur when `withTextPayload = true` (previously extracted for all messages)
   * Added header extraction when `withTextPayload = true`:
      * `message.sender`: Extracted from `From` header
      * `message.recipient`: Extracted from `To` header
      * `message.subject`: Extracted from `Subject` header
   * Added `message.date`: Converts `internalDate` from Unix timestamp to ISO 8601 string format when `withTextPayload = true`

**Why**
* **Enhanced metadata with text payload**: When users enable `withTextPayload` to get email content, they often need corresponding metadata (who sent it, subject line, etc.) to properly process the emails. This change provides that context.
* **Case-insensitive header matching**: Used `toLowerCase()` for header name comparison since email headers are case-insensitive per RFC specifications, ensuring reliable extraction regardless of how Gmail returns header names.
* **LLM-friendly date format**: Converting Unix timestamps to ISO 8601 strings (e.g., "2025-01-15T10:30:00.000Z") makes dates more readable for both humans and LLMs, improving downstream processing.
* **Logical grouping**: Since `withTextPayload` already modifies the message structure for content extraction, it makes sense to include related metadata extraction in the same option.

**Breaking Change**: `message.message_id` is now only extracted when `withTextPayload = true`. Previously it was extracted for all messages. When `withTextPayload = false`, the Message-ID can still be found in `message.payload.headers` array.

**How to test**
**1. Publish the component**:
```bash
pd publish components/gmail/actions/find-email/find-email.mjs
```

**2. Test with `withTextPayload = true`**:
* Run the action on various emails with the option enabled
* Verify that returned messages now include:
  * `message.message_id` (matching the Message-ID header)
  * `message.sender` (matching the From header)
  * `message.recipient` (matching the To header)
  * `message.subject` (matching the Subject header)
  * `message.date` (in ISO 8601 format, e.g., "2025-01-15T10:30:00.000Z")
* Test with emails that have headers in different cases (e.g., "message-id" vs "Message-ID" vs "MESSAGE-ID")

**3. Test with `withTextPayload = false`**:
* Confirm that these new fields are NOT added when the option is disabled
* Verify existing behavior remains unchanged

**4. Edge cases**:
* Test emails with missing headers (fields should be undefined, no errors)
* Test emails with duplicate headers (should use the first occurrence)
* Test with malformed or empty headers
* Test with invalid `internalDate` values

### CODE REVIEW

**Comment by pipedream-component-development** [comment]
Thank you so much for submitting this! We've added it to our backlog to review, and our team has been notified.

**Comment by pipedream-component-development** [comment]
Thanks for submitting this PR! When we review PRs, we follow the [Pipedream component guidelines](https://pipedream.com/docs/components/guidelines/). If you're not familiar, here's a quick checklist:
- [ ] [Create components to address specific use cases whenever possible](https://pipedream.com/docs/components/guidelines/#component-scope)
- [ ] [Component `key`s should follow the format `app_name_slug-slugified-component-name`](https://pipedream.com/docs/components/guidelines/#component-key-pattern)
- [ ] Components should follow the [standard directory structure](https://pipedream.com/docs/components/guidelines/#folder-structure)
- [ ] [Prefer Node.js client libraries to REST APIs](https://pipedream.com/docs/components/guidelines/#using-apis-vs-client-libraries)
- [ ] When making API requests, [handle pagination](https://pipedream.com/docs/components/guidelines/#pagination) to ensure all data / events are processed
- [ ] [Use `secret` props](https://pipedream.com/docs/components/guidelines/#capturing-sensitive-data) to capture sensitive data
- [ ] Props and methods should be defined in [app files](https://pipedream.com/docs/components/guidelines/#app-files) whenever possible
- [ ] Document methods with [JS Docs](https://pipedream.com/docs/components/guidelines/#methods)
- [ ] [Use `optional` props](https://pipedream.com/docs/components/guidelines/#optional-vs-required-props) whenever possible, and [set a `default` value](https://pipedream.com/docs/components/guidelines/#default-values) where you can
- [ ] Use [async options](https://pipedream.com/docs/components/guidelines/#async-options) to accept user input wherever possible

File: components/gmail/actions/find-email/find-email.mjs [hunk 1]
```diff
@@ -60,11 +60,39 @@ export default {
...
+        if (recipientHeader) {
+          message.recipient = recipientHeader.value;
+        }
+
+        const subjectHeader = message.payload?.headers?.find(
+          (h) => h.name.toLowerCase() === "subject",
+        );
+        if (subjectHeader) {
+          message.subject = subjectHeader.value;
+        }
```
Comment (jcortes) [inline components/gmail/actions/find-email/find-email.mjs]: Can you actually leave all this code outside the `if (this.withTextPayload) {` statement?

Comment (hubert-marek) [reply components/gmail/actions/find-email/find-email.mjs]: Done! Originally, I moved the header extraction inside `withTextPayload` to keep the response size smaller when users don't need the text payload. However, I agree that having these metadata fields available for all messages makes the action more useful and easier to work with. Thanks for the review!

**Comment by vunguyenhung** [comment]
Hi everyone, all test cases are passed! Ready for release! 

Test report
https://vunguyenhung.notion.site/Components-Gmail-Update-find-email-action-to-v0-1-2-20fbf548bb5e81c6b3acdeeba737cf38
